.github/workflows/codeql.yaml

```
@@ -0,0 +1,43 @@
+---
+name: codeQL
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+  workflow_dispatch:
+
+jobs:
+  analyze:
+    name: Static analysis with CodeQL
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    env:
+      NEXUS_UNAME: ${{ secrets.NEXUS_USERNAME }}
+      NEXUS_PWD: ${{ secrets.NEXUS_PASSWORD }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Setup Java
+        uses: actions/setup-java@v3
+        with:
+          distribution: "corretto"
+          java-version: "17"
+
+      - name: Setup Gradle
+        uses: gradle/gradle-build-action@v2
+        with:
+          gradle-version: 7.3
```

Even though it doesn't technically change anything, try bumping this action to keep dependency-review happy.

```suggestion
      - name: Setup Gradle
        uses: gradle/gradle-build-action@v2.4.2
        with:
          gradle-version: 7.3
```

Side note: We might need to review the dependency-review configuration used to not block our users. It seems that the configuration at the repository level are not doing anything for this action.